### PR TITLE
Line length fix

### LINKED ISSUE #168: Line length reformatting
Adam was right, it looks like in my past life I turned off the line length checking in the repo. This has resulted in pep8 not checking this.

To remedy this, we'll need to open a branch with the .travis.yml file edited to not ignore 501, and then go in and fix all of the resulting pep8 errors.

### PR DESCRIPTION
**Pull request by ackagel**
Gradually fixing line length errors and doing _*some*_ relevant code/doc-string cleanups as I go.  This won't need to be done again since E501 (max line length) ignoring has been disabled in `pytest.ini`.  

Since there's a lot to go through, I'm gonna break it up into several smaller review requests.

Closes #168

### CODE REVIEW

**Comment by ngreenwald** [comment]
Let's wait until @alex-l-kong finishes his PR before doing any more work on this branch.

**Comment by ackagel** [comment]
still have to fix some in `utils`, but just wanted a quick yay/nay on `segmentation` fixes

**Comment by ackagel** [comment]
okeedokee.  I did it that way since google's style guide says either way is good

**Comment by ngreenwald** [comment]
Okay, I stand corrected

**Comment by ngreenwald** [comment]
If the bible says it's okay then it's okay

File: ark/analysis/spatial_analysis.py [hunk 1]
```diff
@@ -109,27 +115,38 @@ def calculate_cluster_spatial_enrichment(all_data, dist_matrices_dict, included_
...
+
     Returns:
-        values (list): a list with each element consisting of a tuple of
-            closenum and closenumrand for each point included in the analysis
-        stats (xarray): an Xarray with dimensions (fovs, stats, number of channels, number of channels) The included stats
-            variables are:
-            z, muhat, sigmahat, p, h, adj_p, and cluster_names for each point in the analysis"""
+         - a list with each element consisting of a tuple of closenum and closenumrand for each
+         point included in the analysis
+         - an Xarray with dimensions (fovs, stats, number of channels, number of channels). The
```
Comment (ngreenwald) [inline ark/analysis/spatial_analysis.py:146]: I think we can leave xarray lowercase? Probably we would say pandas array, not Pandas array. But I could be convinced.

File: ark/analysis/visualize.py [hunk 2]
```diff
@@ -71,7 +99,14 @@ def visualize_patient_population_distribution(cell_data, patient_col_name, popul
 
     # Plot by count
     if show_distribution:
-        get_sorted_data(cell_data, patient_col_name, population_col_name).plot.bar(stacked=True, colormap=color_map)
+        get_sorted_data(
+            cell_data,
+            patient_col_name,
+            population_col_name
```
Comment (ngreenwald) [inline ark/analysis/visualize.py:105]: I don't love this, seems a bit hard to read

File: ark/segmentation/marker_quantification.py [hunk 3]
```diff
@@ -108,14 +111,17 @@ def generate_expression_matrix(segmentation_labels, image_data, nuclear_counts=F
...
+            FOV, potentially across multiple cell compartments
+        image_data (xarray.DataArray):
+            xarray containing all of the channel data across all FOVs
+        nuclear_counts (bool):
+            boolean flag to determine whether nuclear counts are returned
 
     Returns:
-        normalized_data (pandas): marker counts per cell normalized by cell size
-        arcsinh_data (pandas): arcsinh transfomed marker counts per cell normalized by cell size
+        Tuple of marker counts per cell normalized and arcsinh transfomed marker counts per cell
```
Comment (ngreenwald) [inline ark/segmentation/marker_quantification.py:123]: This is a bit hard to follow. I wonder if there's a better way we can describe tuple return data?

File: ark/utils/data_utils_test.py [hunk 4]
```diff
@@ -512,7 +512,14 @@ def test_split_img_stack():
         junk_img_chan_last = np.zeros((1024, 1024, 10))
         io.imsave(os.path.join(stack_dir, "channel_data.tif"), junk_img_chan_last)
 
-        data_utils.split_img_stack(stack_dir, output_dir, stack_list, indices, names, channels_first=False)
+        data_utils.split_img_stack(
+            stack_dir,
```
Comment (ngreenwald) [inline ark/utils/data_utils_test.py:516]: I think for stuff that's this short, indenting every line just makes the code take up more space and isn't that helpful

File: ark/utils/data_utils_test.py [hunk 5]
```diff
@@ -527,10 +534,17 @@ def test_split_img_stack():
...
         io.imsave(os.path.join(stack_dir, "channel_data.tif"), junk_img_chan_first)
 
-        # clear the original channel_data directory so an error doesn't get thrown trying to recreate it
+        # clear the original channel_data directory
         rmtree(os.path.join(output_dir, "channel_data"))
 
-        data_utils.split_img_stack(stack_dir, output_dir, stack_list, indices, names, channels_first=True)
+        data_utils.split_img_stack(
+            stack_dir,
+            output_dir,
```
Comment (ngreenwald) [inline ark/utils/data_utils_test.py:542]: Same here

File: ark/utils/spatial_analysis_utils.py [hunk 6]
```diff
@@ -192,21 +221,20 @@ def calculate_enrichment_stats(close_num, close_num_rand):
...
-        sigmahat (np array): predicted standard deviation values of close_num_rand
-            random distribution
-        p (np array): p values for corresponding markers, for both positive
-            and negative enrichment
-        h (np array): matrix indicating whether
-            corresponding marker interactions are significant
-        adj_p (np array): fdh_br adjusted p values"""
+        xarray contining the following statistics for marker to marker enrichment
+            - z: z scores for corresponding markers
+            - muhat: predicted mean values of close_num_rand random distribution
```
Comment (ngreenwald) [inline ark/utils/spatial_analysis_utils.py:232]: This is really nice

File: ark/utils/synthetic_spatial_datagen.py [hunk 7]
```diff
@@ -283,8 +330,12 @@ def generate_two_cell_test_nuclear_signal(segmentation_mask, cell_centers,
 
         # generate the nuclear region in the middle of the cell with the same cell center
         # and set signal to a uniform value
-        nuc_region_x, nuc_region_y = circle(center[0], center[1], nuc_radius + nuc_uncertainty_length,
-                                            shape=size_img)
+        nuc_region_x, nuc_region_y = circle(
+            center[0],
+            center[1],
```
Comment (ngreenwald) [inline ark/utils/synthetic_spatial_datagen.py:335]: This one as well, this could be two lines instead

File: ark/utils/synthetic_spatial_datagen.py [hunk 8]
```diff
@@ -337,50 +404,71 @@ def generate_two_cell_test_membrane_signal(segmentation_mask, cell_centers,
...
-                                                                    cell_radius=cell_radius,
-                                                                    memb_cell_ids=memb_cell_ids,
-                                                                    memb_thickness=memb_thickness,
-                                                                    memb_signal_strength=memb_signal_strength,
-                                                                    memb_uncertainty_length=memb_uncertainty_length)
+    sample_nuclear_signal = generate_two_cell_test_nuclear_signal(
+        segmentation_mask=sample_segmentation_mask,
+        cell_centers=sample_cell_centers,
+        size_img=size_img,
+        nuc_cell_ids=nuc_cell_ids,
```
Comment (ngreenwald) [inline ark/utils/synthetic_spatial_datagen.py:458]: These ones it makes sense, since the functions starts so far right already

File: tox.ini [hunk 9]
```diff
@@ -1,16 +1,15 @@
 # Configuration of py.test
 [pytest]
 addopts=-v
+        -s
```
Comment (ngreenwald) [inline tox.ini:4]: What's the rationale for switching the file name?

Comment (ackagel) [reply tox.ini:4]: pycodestyle explicitly needs a setup.cfg (depreciated) or a tox.ini for adding any settings (like max line length), and pytest will gladly take either (we could also have a pytest.ini __and__ a tox.ini, but I figured they might as well be in the same file)

also, the added `-s` flag shows the rationale for skipped warnings.  I added it because pycodestyle will skip unchanged, previously passed files for style-review (it's safe in general, but can be confusing for local testing w/o the `-s` flag active.  tbh, I'm not a big fan of it, but there's no (well-documented) way to turn it off)

**Comment by ackagel** [comment]
I live for the `git add *` dream

My `.gitignore` will be long enough to be the next Harry Potter novel

but srsly my b

**Comment by ngreenwald** [comment]
Hah, it's all good, it's the same file that's already there

File: ark/analysis/dimensionality_reduction.py [hunk 10]
```diff
@@ -9,20 +7,33 @@
...
+        color_map (str):
+            Name of MatPlotLib ColorMap used, default is Spectral
+        algorithm (str):
+            Name of dimensionality reduction algorithm, default is UMAP
+        save_dir (str):
+            Directory to save plots, default is None
+
+    Returns:
+        None
+
```
Comment (ngreenwald) [inline ark/analysis/dimensionality_reduction.py:30]: extra space

File: ark/analysis/spatial_analysis.py [hunk 11]
```diff
@@ -7,32 +7,41 @@
...
+            column with the cell fovs. Default is 'SampleID'
 
     Returns:
-        values (list): a list with each element consisting of a tuple of
-            closenum and closenumrand for each point included in the analysis
-        stats (xarray): an Xarray with dimensions (fovs, stats, number of channels, number of channels) The included stats
-            variables are:
-            z, muhat, sigmahat, p, h, adj_p, and cluster_names for each point in the analysis"""
+        tuple (list, xarray.DataArray):
+            Two element tuple containing
```
Comment (ngreenwald) [inline ark/analysis/spatial_analysis.py:38]: I think this line is redundant. The one above is super clear

File: ark/analysis/spatial_analysis.py [hunk 12]
```diff
@@ -109,27 +118,40 @@ def calculate_cluster_spatial_enrichment(all_data, dist_matrices_dict, included_
...
+            column with the cell labels. Default is 'cellLabelInImage'
+
     Returns:
-        values (list): a list with each element consisting of a tuple of
-            closenum and closenumrand for each point included in the analysis
-        stats (xarray): an Xarray with dimensions (fovs, stats, number of channels, number of channels) The included stats
-            variables are:
-            z, muhat, sigmahat, p, h, adj_p, and cluster_names for each point in the analysis"""
+        tuple (list, xarray.DataArray):
+            Two element tuple containing
```
Comment (ngreenwald) [inline ark/analysis/spatial_analysis.py:148]: Same. I really like this new format

File: ark/analysis/spatial_analysis.py [hunk 13]
```diff
@@ -183,26 +206,35 @@ def calculate_cluster_spatial_enrichment(all_data, dist_matrices_dict, included_
...
+            column with the cell fovs. Default is 'SampleID'
+        cluster_id_col (str):
+            column with the cell phenotype IDs. Default is 'FlowSOM_ID'
+        cell_label_col (str):
+            column with the cell labels. Default is 'cellLabelInImage'
+        cluster_name_col (str):
+            column with the cell types. Default is 'cell_type'
+
+    Returns:
+        pandas.DataFrame:
```
Comment (ngreenwald) [inline ark/analysis/spatial_analysis.py:234]: This too, great change

File: ark/analysis/visualize.py [hunk 14]
```diff
@@ -19,44 +25,74 @@ def visualize_z_scores(z, pheno_titles):
...
+            Boolean specifying whether to show graph of total population count, default is true
+        show_distribution (bool):
+            Boolean specifying whether to show graph of population distribution, default is true
+        show_proportion (bool):
+            Boolean specifying whether to show graph of total count, default is true
+        save_dir (str):
+            Directory to save plots, default is None
+
+    Returns:
+        None
```
Comment (ngreenwald) [inline ark/analysis/visualize.py:94]: Do you prefer listing Returns None to just not having it here at all?

File: ark/segmentation/marker_quantification.py [hunk 15]
```diff
@@ -108,14 +112,19 @@ def generate_expression_matrix(segmentation_labels, image_data, nuclear_counts=F
...
+        image_data (xarray.DataArray):
+            xarray containing all of the channel data across all FOVs
+        nuclear_counts (bool):
+            boolean flag to determine whether nuclear counts are returned
 
     Returns:
-        normalized_data (pandas): marker counts per cell normalized by cell size
-        arcsinh_data (pandas): arcsinh transfomed marker counts per cell normalized by cell size
+        tuple (pandas.DataFrame, pandas.DataFrame):
+            Two element tuple containing
```
Comment (ngreenwald) [inline ark/segmentation/marker_quantification.py:125]: Same here, just 
```
tuple(type, type)
     element one description
     element two description
```

File: ark/utils/synthetic_spatial_datagen.py [hunk 16]
```diff
@@ -337,50 +394,72 @@ def generate_two_cell_test_membrane_signal(segmentation_mask, cell_centers,
...
+            will extend nuc_radius by specified length
+        memb_uncertainty_length (int):
+            will extend memb_radius by specified length
 
     Returns:
-        sample_segmentation_mask (numpy): an array which contains the labeled cell regions
-        sample_nuclear_signal (numpy): defines the nuclear signal for the desired cells
-        sample_membrane_signal (numpy): defines the membrane signal for the desired cells
+        tuple (numpy.ndarray, numpy.ndarray, numpy.ndarray):
+            Three element tuple containing
```
Comment (ngreenwald) [inline ark/utils/synthetic_spatial_datagen.py:432]: This one too

File: ark/analysis/visualize.py (hunk 14 above)
Comment (ackagel) [reply ark/analysis/visualize.py:94]: I don't think I have a preference.  Both are fine.  Figure'd it'd be easier to clean up `Returns: None`, than to add them in later.

Comment (ackagel) [reply ark/analysis/visualize.py:94]: it's more common to have nothing there, so I'll go with that
